packages/medusa/src/services/order.js

```diff
@@ -0,0 +1,447 @@
+   * Throws if payment process has been initiated.
+   * @param {string} orderId - id of order to cancel.
+   * @return {Promise} result of the update operation.
+   */
+  async cancel(orderId) {
+    const order = await this.retrieve(orderId)
+
+    if (order.fulfillment_status !== "not_fulfilled") {
+      throw new MedusaError(
+        MedusaError.Types.NOT_ALLOWED,
+        "Can't cancel a fulfilled order"
+      )
+    }
+
+    if (order.payment_status !== "awaiting") {
+      throw new MedusaError(
+        MedusaError.Types.NOT_ALLOWED,
+        "Can't cancel an order with payment processed"
+      )
+    }
+
+    // cancel payment method
+
+    return this.orderModel_.updateOne(
+      {
+        _id: orderId,
+      },
+      {
+        $set: { status: "cancelled" },
+      }
+    )
+  }
+
+  /**
+   * Captures payment for an order.
+   * @param {string} orderId - id of order to capture payment for.
+   * @return {Promise} result of the update operation.
+   */
+  async capturePayment(orderId) {
+    const order = await this.retrieve(orderId)
+
+    if (order.payment_status !== "awaiting") {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "Payment already captured"
+      )
+    }
+
+    const { provider_id, data } = order.payment_method
+    const paymentProvider = await this.paymentProviderService_.retrieveProvider(
+      provider_id
+    )
+
+    await paymentProvider.capturePayment(data)
+
+    return this.orderModel_.updateOne(
+      {
+        _id: orderId,
+      },
+      {
+        $set: { payment_status: "captured" },
+      }
+    )
+  }
+
+  /**
+   * Creates fulfillments for an order.
+   * In a situation where the order has more than one shipping method,
+   * we need to partition the order items, such that they can be sent
+   * to their respective fulfillment provider.
+   * @param {string} orderId - id of order to cancel.
+   * @return {Promise} result of the update operation.
+   */
+  async createFulfillment(orderId) {
+    const order = await this.retrieve(orderId)
+
+    if (order.fulfillment_status !== "not_fulfilled") {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "Order is already fulfilled"
+      )
+    }
+
+    const { shipping_methods, items } = order
+
+    // if only one shipping method is attached to the order, we simply create
+    // the order for the single provider given in the method
+    if (shipping_methods.length === 1) {
+      const fulfillmentProviderId = shipping_methods[0].provider_id
+      const fulfillmentProvider = this.fulfillmentProviderService_.retrieveProvider(
+        fulfillmentProviderId
+      )
+
+      shipping_methods[0].items = items
+
+      await fulfillmentProvider.createOrder(shipping_methods[0].data, items)
+
+      return this.orderModel_.updateOne(
+        {
+          _id: orderId,
+        },
+        {
+          $set: { fulfillment_status: "fulfilled" },
+        }
+      )
+    }
+
```

Comment:
I think we should refactor this a bit - I get that we don't want to go through all items if we only have 1 `shipping_method`, however, this makes the partitioning easier so we could easily change `partitionItems_` to avoid the unnecessary iterations. 

In `partitionItems_` do this:

```js
    let updatedMethods = []
    // partition order items to their dedicated shipping method
    await Promise.all(
      shipping_methods.map(async method => {
        const { profile_id } = method
        const profile = await this.shippingProfileService_.retrieve(profile_id)
        // for each method find the items in the order, that are associated
        // with the profile on the current shipping method
        if (shipping_methods.length === 1) { // <-------- THIS
           method.items = items 
        } else {
          method.items = items.filter(({ content }) => {
            if (Array.isArray(content)) {
              // we require bundles to have same shipping method, therefore:
              return profile.products.includes(content[0].product._id)
            } else {
              return profile.products.includes(content.product._id)
            }
          })
        }
        updatedMethods.push(method)
      })
    )
    return updatedMethods
```


Doing this, our map iteration on line 329 will just go through one item in the `shipping_methods` array which is no problem :D